lib/rubocop/cop/style/format_string.rb

```diff
@@ -40,6 +40,55 @@ def message(detected_style)
         def method_name(style_name)
           style_name == :percent ? 'String#%' : style_name
         end
+
+        def autocorrect(node)
+          lambda do |corrector|
+            detected_method = node.loc.selector.source.to_sym
+
+            case detected_method
+            when :%
+              autocorrect_from_percent(corrector, node)
+            when :format, :sprintf
+              case style
+              when :percent
+                autocorrect_to_percent(corrector, node)
+              when :format, :sprintf
+                corrector.replace(node.loc.selector, style.to_s)
+              end
+            end
+          end
+        end
+
+        private
+
+        def autocorrect_from_percent(corrector, node)
+          format, *_, args = *node
```

Comment:
I am sorry my response is late.

As you said, that is not a pure String.

```ruby
[1] pry(#<RuboCop::Cop::Style::FormatString>)> node
=> s(:send,
  s(:str, "%d"), :%,
  s(:int, 10))
[2] pry(#<RuboCop::Cop::Style::FormatString>)> format, *_, args = *node
=> [s(:str, "%d"), :%, s(:int, 10)]
[3] pry(#<RuboCop::Cop::Style::FormatString>)> format
=> s(:str, "%d")
[4] pry(#<RuboCop::Cop::Style::FormatString>)> format.class
=> RuboCop::AST::Node
```

That is a node object which represents a String.

`receiver`, might be suitable as local var name. b535e9e

There is a similar expression in this project.

ex. https://github.com/bbatsov/rubocop/blob/v0.48.1/lib/rubocop/cop/rails/dynamic_find_by.rb#L46